python/lsst/pipe/base/pipelineTask.py

```diff
@@ -320,6 +354,37 @@ def getDatasetTypes(cls, config, configClass):
                 dsTypes[key] = DatasetTypeDescriptor.fromConfig(value)
         return dsTypes
 
+    @classmethod
+    def getPerDatasetTypeDimensions(cls, config):
+        """Return any Dimensions that are permitted to have different values
+        for different DatasetTypes within the same quantum.
+
+        Parameters
+        ----------
+        config : `Config`
+            Configuration for this task.
+
+        Returns
+        -------
+        dimensions : `~collections.abc.Set~ of `Dimension` or `str`
+            The dimensions or names thereof that should be considered
+            per-DatasetType.
+
+        Notes
+        -----
+        Any Dimension declared to be per-DatasetType by a PipelineTask must
+        also be declared to be per-DatasetType by other PipelineTasks in the
+        same Pipeline.
+
+        The classic example of a per-DatasetType dimension is the
+        ``CalibrationLabel`` dimension that maps to a validity range for
+        master calibrations.  When running Instrument Signature Removal, one
+        does not care that different dataset types like the flat, bias, dark,
+        etc. have different validity ranges, as long as those validity ranges
```

Comment:
Pedantic comment but please drop the `etc.` and say "types like the flat, bias, or dark" -- the "like" already implies you are giving examples.